query-engine/driver-adapters/js/connector-test-kit-executor/src/index.ts

```diff
@@ -170,10 +181,39 @@ function respondOk(requestId: number, payload: unknown) {
 }
 
 async function initQe(url: string, prismaSchema: string, logCallback: qe.QueryLogCallback): Promise<[engines.QueryEngineInstance, ErrorCapturingDriverAdapter]> {
-    const pool = new pgDriver.Pool({ connectionString: url })
-    const adapter = bindAdapter(new pg.PrismaPg(pool))
-    const engineInstance = qe.initQueryEngine(adapter, prismaSchema, logCallback)
-    return [engineInstance, adapter];
+    const adapter = await adapterFromEnv(url) as DriverAdapter
+    const errorCapturingAdapter = bindAdapter(adapter)
+    const engineInstance = qe.initQueryEngine(errorCapturingAdapter, prismaSchema, logCallback)
+    return [engineInstance, errorCapturingAdapter];
+}
+
+async function adapterFromEnv(url: string): Promise<DriverAdapter> {
+    const adapter = process.env.DRIVER_ADAPTER ?? ''
+
+    if (adapter == '') {
+        throw new Error("DRIVER_ADAPTER is not defined or empty.")
+    }
+
+    if (!(adapter in SUPPORTED_ADAPTERS)) {
+        throw new Error(`Unsupported driver adapter: ${adapter}`)
+    }
+
+    return await SUPPORTED_ADAPTERS[adapter](url);
+}
+
+async function pgAdapter(url: string): Promise<DriverAdapter> {
+    const pool = new pgDriver.Pool({connectionString: url})
+    return new prismaPg.PrismaPg(pool)
+}
+
+async function neonAdapter(_: string): Promise<DriverAdapter> {
+    const connectionString = process.env.DRIVER_ADAPTER_URL_OVERRIDE ?? ''
```

Comment:
Not blocking in any way and can be addressed later if we decide to change it, but I wonder if it would actually be better to have separate environment variables (e.g. `DRIVER_ADAPTER_NEON_URL` etc) for each adapter, if the current `DRIVER_ADAPTER_URL_OVERRIDE` is not being read in any centralised way anyway and each adapter constructor function is responsible for reading it on its own.

Some reasoning behind it:
* It's not always just one variable per adapter, some adapters have none (pg), some have two or three (Turso).
* It would be possible to set all of them in `.envrc.local` once and not comment/uncomment stuff or source separate files to switch between adapters.

Reply:
When I designed this I opted for the opposite reasoning. Which is, whatever you need is by an adapter, if any, is in that variable. For instance, if it has 3 urls, it can be coalesced into a single env var, joined by a separator.

The reason for that choice is that you don't need to remember which is the variables per connector for tests. This behavior is symmetric with how we test other drivers: ex. we use a single URL variable for all connectors, and then we use a variable connector.

I see smoke tests take the opposite approach, the one you mentioned. I think we should aim at consistency, and have something akin to what we have when testing a rust connector, which is a set of variables being set accordingly, when we issue a `make dev-*` command. Rather than having all the options in different variables in the .envrc.

I'm open to discussing this further async.